### ISSUE TITLE
Image Rendering Issues since upgrading to 1.4.0

### ISSUE THREAD

**Issue by stuieordie** [pelican-plugins/photos#62]
I'm using version 1.4.0 from pip installation. 

A lot of the photos in my gallery aren't getting rendered and a lot of errors display when running `pelican -r -l`. I'm getting the wrong format warning for blacklist.txt but it still appears to be omitting the listed images. Note that some images do get created but others do not. I tried setting the `PHOTO_EXIF_KEEP` to false and some images still don't get created.

```
[11:03:48] WARNING  Wrong format in file /home/mstuart/Pictures/Personal/Blog/20220907/blacklist.txt on line 1                                                                log.py:91
           WARNING  Wrong format in file /home/mstuart/Pictures/Personal/Blog/20220907/blacklist.txt on line 2                                                                log.py:91
           WARNING  Wrong format in file /home/mstuart/Pictures/Personal/Blog/20220907/blacklist.txt on line 3                                                                log.py:91
           WARNING  Wrong format in file /home/mstuart/Pictures/Personal/Blog/20220907/blacklist.txt on line 4                                                                log.py:91
           WARNING  Wrong format in file /home/mstuart/Pictures/Personal/Blog/20220907/blacklist.txt on line 5                                                                log.py:91
           WARNING  Wrong format in file /home/mstuart/Pictures/Personal/Blog/20220907/blacklist.txt on line 6                                                                log.py:91
           ERROR    photos: 'GPS'                                                                                                                                             log.py:96
           WARNING  photos: An exception occurred                                                                                                                             log.py:91
                    multiprocessing.pool.RemoteTraceback:                                                                                                                              
                    """                                                                                                                                                                
                    Traceback (most recent call last):                                                                                                                                 
                      File "/usr/lib/python3.10/multiprocessing/pool.py", line 125, in worker                                                                                          
                        result = (True, func(*args, **kwds))                                                                                                                           
                      File "/home/mstuart/.local/lib/python3.10/site-packages/pelican/plugins/photos/photos.py", line 1053, in process                                                 
                        image = advanced_func(image, self, *operation_args, **operation_kwargs)                                                                                        
                      File "/home/mstuart/.local/lib/python3.10/site-packages/pelican/plugins/photos/photos.py", line 1147, in _operation_manipulate_exif                              
                        image_meta.exif_result.pop("GPS")                                                                                                                              
                    KeyError: 'GPS'                                                                                                                                                    
                    """                                                                                                                                                                
                                                                                                                                                                                       
                    The above exception was the direct cause of the following exception:                                                                                               
                                                                                                                                                                                       
                    KeyError: 'GPS'                                                                                                                                                    
           WARNING  Empty alt attribute for image 20220831_195729.jpg in /home/mstuart/Coding/Home/Blog/Pelican/blog.stuieordie.net/content/mt-titcomb-farmington.md          log.py:91
           ERROR    photos: "dump" got wrong type of exif value.                                                                                                              log.py:96
                    41729 in Exif IFD. Got as <class 'int'>.                                                                                                                           
           ERROR    photos: "dump" got wrong type of exif value.                                                                                                              log.py:96
                    41729 in Exif IFD. Got as <class 'int'>.                                                                                                                           
           ERROR    photos: "dump" got wrong type of exif value.                                                                                                              log.py:96
                    41729 in Exif IFD. Got as <class 'int'>.                                                                                                                           
           ERROR    photos: Given thumbnail is too large. max 64kB                                                                                                            log.py:96
           ERROR    photos: Given thumbnail is too large. max 64kB                                                                                                            log.py:96
Suspension not allowed here
           ERROR    photos: encoder error -2 when writing image file                                                                                                          log.py:96
Suspension not allowed here
[11:03:49] ERROR    photos: encoder error -2 when writing image file                                                                                                          log.py:96
           ERROR    photos: "dump" got wrong type of exif value.                                                                                                              log.py:96
                    41729 in Exif IFD. Got as <class 'int'>.                                                                                                                           
           ERROR    photos: "dump" got wrong type of exif value.                                                                                                              log.py:96
                    41729 in Exif IFD. Got as <class 'int'>.                                                                                                                           
           ERROR    photos: "dump" got wrong type of exif value.                                                                                                              log.py:96
                    41729 in Exif IFD. Got as <class 'int'>.                                                                                                                           
           ERROR    photos: "dump" got wrong type of exif value.                                                                                                              log.py:96
                    41729 in Exif IFD. Got as <class 'int'>.                                                                                                                           
Suspension not allowed here
           ERROR    photos: encoder error -2 when writing image file                                                                                                          log.py:96
Suspension not allowed here
           ERROR    photos: encoder error -2 when writing image file                                                                                                          log.py:96
Suspension not allowed here
           ERROR    photos: encoder error -2 when writing image file                                                                                                          log.py:96
Suspension not allowed here
           ERROR    photos: encoder error -2 when writing image file                                                                                                          log.py:96
Suspension not allowed here
           ERROR    photos: encoder error -2 when writing image file                                                                                                          log.py:96
Suspension not allowed here
           ERROR    photos: encoder error -2 when writing image file                                                                                                          log.py:96
[11:03:50] ERROR    photos: 'GPS'                                                                                                                                             log.py:96
           ERROR    photos: 'GPS'                                                                                                                                             log.py:96
Suspension not allowed here
           ERROR    photos: encoder error -2 when writing image file                                                                                                          log.py:96
Suspension not allowed here
           ERROR    photos: encoder error -2 when writing image file                                                                                                          log.py:96
Suspension not allowed here
           ERROR    photos: encoder error -2 when writing image file                                                                                                          log.py:96
           ERROR    photos: "dump" got wrong type of exif value.                                                                                                              log.py:96
                    41729 in Exif IFD. Got as <class 'int'>.                                                                                                                           
Suspension not allowed here
           ERROR    photos: encoder error -2 when writing image file                                                                                                          log.py:96
           ERROR    photos: "dump" got wrong type of exif value.                                                                                                              log.py:96
                    41729 in Exif IFD. Got as <class 'int'>.                                                                                                                           
           ERROR    photos: "dump" got wrong type of exif value.                                                                                                              log.py:96
                    41729 in Exif IFD. Got as <class 'int'>.                                                                                                                           
           ERROR    photos: "dump" got wrong type of exif value.                                                                                                              log.py:96
                    41729 in Exif IFD. Got as <class 'int'>.                                    
```

**Comment by phibos** [comment]
Thanks for reporting the issue. Is the log from ```PHOTO_EXIF_KEEP=True``` or from ```PHOTO_EXIF_KEEP=False```. Can you please provide a list of the installed packages ```pip list```, at least of ```pelican```, ```pillow``` and ```piexif```.

**Comment by stuieordie** [comment]
```
$ python3 -m pip list | grep -e "pillow\|piexif\|pelican"
pelican                 4.8.0
pelican-photos          1.4.0
piexif                  1.1.3
```
Not sure why pillow doesn't show up in the list but it is installed
```
$ python3 -m pip install pillow
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: pillow in /home/mstuart/.local/lib/python3.10/site-packages (8.4.0)
```

**Comment by phibos** [comment]
Looks like some of the issues are related to piexif. In version < 1.4.0 piexif is included but not used.

```
"dump" got wrong type of exif value.
41729 in Exif IFD. Got as <class 'int'>
```
See: hMatoba/Piexif#95 open since 2019

```
Given thumbnail is too large. max 64kB 
```
https://github.com/hMatoba/piexifjs/blob/7b9140ab8ebb8ff620bb20f6319a337dd150092b/piexif.js#L236-L243


At the moment I was unable to reproduce the issues with the images I use. Could you please provide a raw image to debug the issues?

**Comment by phibos** [comment]
I have created a new issue #63 to fix the issue with the blacklist.txt.

**Comment by stuieordie** [comment]
Yes no problem. Here is one of the images.. I can provide more if needed.

[20220903_080132](https://user-images.githubusercontent.com/25537622/190281173-3b1b094e-d266-446a-b1be-6741ae8bd3c8.jpg)


Current package list again
```
$ python3 -m pip list | grep -ie "pillow\|piexif\|pelican"
pelican                 4.8.0
pelican-photos          1.4.0
piexif                  1.1.3
Pillow                  9.2.0
```

**Comment by Jeffmackinnon** [comment]
Is there a fix for this bug? I haven't been able to update my site for a while since it keeps crashing on 'GPS' if I ask it to remove GPS information or '0th' when I try leaving it.

Thanks, 
Jeff M

**Comment by phibos** [comment]
@Jeffmackinnon multiple issues have been mixed up in this report. So what issue do you mean. Can you provide the part of the log you are talking about. If it is possible it would be nice if you could run pelican with ```--debug``` to get more information and if you could provide a sample image. It is okay if you don't want to share your raw images with the GPS data, we can start with the debug output.

**Comment by phibos** [comment]
The remaining issues should have been fixed with PR #76. All the changes have been merged into the main branch and will be in the next release. Please feel free to test.

**Comment by justinmayer** [comment]
This issue should hopefully be addressed by the just-released [Photos 1.5.0](https://github.com/pelican-plugins/photos/releases/tag/1.5.0). If not, feel free to comment on this issue so we can re-open it.
